### PR TITLE
[CI-3858] Resolve body with `task_id` on mutating items and variations functions

### PR DESCRIPTION
**Pull request by ZSnake**
- Update the following mutating functions to return the API call response body instead of resolving an empty promise:
  - `createOrReplaceItems`
  - `updateItems`
  - `deleteItems`
  - `createOrReplaceVariations`
  - `updateVariations`
  - `deleteVariations`
- Fix broken tests related to `getBrowseResults` and `Facet Configurations` cleanup hook

### CODE REVIEW

File: src/modules/catalog.js
Comment (Mudaafi) [inline src/modules/catalog.js:258]: Lets move this within the conditional `if (response.ok) {`. If we return an error, it might not be in json so we'll resolve a json-parse error instead of the http error on line 264. Besides, that's the only place we're using `body` anyways.

**Review by Mudaafi — CHANGES_REQUESTED** [review]
Let's also update the types in `catalog.d.ts`

**Review by Mudaafi — APPROVED** [review]
LGTM. let's get a second review since this touches a lot of methods

**Review by stanlp1 — APPROVED** [review]
LGTM

**Pull request merged**
